examples/buildkit1/buildkit.go

```diff
@@ -88,7 +89,14 @@ func goFromGit(repo, tag string) llb.StateOption {
 		Dirf("/go/src/%s", repo).
 		Run(llb.Shlexf("git checkout -q %s", tag)).Root()
 	return func(s llb.State) llb.State {
-		return s.With(copyFrom(src, "/go", "/")).Reset(s).Dir(src.GetDir())
+		return s.With(copyFrom(src, "/go", "/")).Reset(s).Async(func(ctx context.Context, s llb.State) (llb.State, error) {
+			// TODO: add s.With(s2.DirValue) or s.With(llb.Dir(s2)) or s.Reset(s2, llb.DirMask)?
```

Comment:
Out of scope for this PR?

Reply:
Yes. You can comment on which one you prefer though.

Reply:
Prefer `s.With(llb.Dir(s2))`

Reply:
Per #1429, need a new solution for this.